x/ecocredit/server/core/features/msg_create_batch.feature

```diff
@@ -408,3 +408,106 @@ Feature: Msg/CreateBatch
       """
 
     # no failing scenario - response should always be empty when message execution fails
+
+  Rule: Events are emitted
+
+    Background:
+      Given a credit type with abbreviation "C"
+      And ecocredit module's address "regen15406g34dl5v9780tx2q3vtjdpkdgq4hhegdtm9"
+      And a credit class with class id "C01" and issuer alice
+      And a project with project id "C01-001"
+
+    Scenario:  Events EventRetire, EventMint, EventTransfer, and EventCreateBatch are emitted
+      When creates a batch from project "C01-001" and issues "10" retired credits to "regen1sl2dsfyf2znn48ehwqg28cv3nuglxkx4h7q5l8" from "US-WA"
+      Then expect event retire with properties
+      """
+      {
+        "owner": "regen1sl2dsfyf2znn48ehwqg28cv3nuglxkx4h7q5l8",
+        "batch_denom": "C01-001-20200101-20210101-001",
+        "amount": "10",
+        "jurisdiction": "US-WA"
+      }
+      """
```

Comment:
According to the BDD books, it's best practice to limit the number of items you are checking in a scenario. In this case, it might be best to break each event out into a separate scenario.

The integration test example I shared is a bit different and what is referred to as a "journey" in the BDD books, but for unit tests, it is recommended to reduce each scenario to the smallest unit.

Ok to leave as is for now but we should further discuss when writing guidelines.

Reply:
https://github.com/regen-network/regen-ledger/issues/1242#issuecomment-1225190275

Reply:
[refactor: use one check per scenario](https://github.com/regen-network/regen-ledger/pull/1394/commits/93d298706c461aafff0c29c176ccb2301c08e3f0)